OpenRA.Mods.Common/Orders/EnterAlliedActorTargeter.cs

```diff
@@ -16,11 +16,11 @@ namespace OpenRA.Mods.Common.Orders
 {
 	public class EnterAlliedActorTargeter<T> : UnitOrderTargeter where T : ITraitInfo
 	{
-		readonly Func<Actor, bool> canTarget;
+		readonly Func<Actor, TargetModifiers, bool> canTarget;
 		readonly Func<Actor, bool> useEnterCursor;
 
 		public EnterAlliedActorTargeter(string order, int priority,
-			Func<Actor, bool> canTarget, Func<Actor, bool> useEnterCursor)
+			Func<Actor, TargetModifiers, bool> canTarget, Func<Actor, bool> useEnterCursor)
```

Comment:
#16607 solves the same use case by adding a `Func<bool> requireForceMove` argument.
I like that this approach doesn't hardcode a new concept with the targeter, but it has the downside of requiring shims in `Repairable` to translate.

Do you have a strong preference or thoughts on either of the two approaches?

Reply:
Like you said, either approach has downsides and upsides so I don't have strong preference either way. That said, an added benefit of this approach is that it works equally for any modifier key, not just `ForceMove`. That makes this a really generic approach that could potentially be repurposed for many different concepts. Considering this, I have a slight preference for this approach.

Reply:
Changed #16607 to use this approach. If we merge that first it will reduce the diff here a bit.